src/components/pill/pill.style.js

```diff
@@ -12,26 +13,25 @@ import OptionsHelper from '../../utils/helpers/options-helper';
 const setTheme = (theme) => {
   switch (theme.name) {
     case THEMES.classic:
-    case THEMES.medium:
-    case THEMES.large:
     case THEMES.base:
-      return THEMES.small;
+      return smallTheme;
 
     default:
-      return theme.name;
+      return theme;
   }
 };
 
 const PillStyle = styled.span`
  ${({
-    colourVariant, theme, inFill, isDeletable
+    colourVariant, theme, inFill, isDeletable, role
   }) => {
     const { colors } = baseTheme;
-    const themeName = setTheme(theme);
-    const styleSet = styleConfig[themeName];
+    const correctedTheme = setTheme(theme);
+    const styleSet = styleConfig(correctedTheme)[role];
+    const colour = (role === 'status') ? colourVariant : 'primary';
```

Comment:
We should probably change `colour` and `colourVariant` to Americanised spellings, for consistency with the rest of the code-base